### ISSUE TITLE
[REQ, FEAT] A Maintenance mode window / 'calendar' per device would be nice

### ISSUE THREAD

**Issue by jaxjexjox** [louislam/uptime-kuma#4018]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

If a system regularly goes down from say 09:00pm until 11pm, every second Saturday, it would be lovely to be able to click a box, go into maintenance calendar designer and just mark out rules which stop alerting for this system during this time window.

### ✔️ Solution

Option 1:
Per 'system' calendar, that allows for a 'time window' where the system being out of action, doesn't trigger a notification.

Option 2:
**Another solution that would also be perhaps less work for the developers.......** 

Configuring an option to disable *ALL* alerting outside of XYZ hours (example 7am -> 8pm Monday to Friday) Alerting is enabled
Any outages NOT during this time, do not get notification.

Example:
3am Sunday night, System 1,2,3 go down.
7am Monday morning, systems 1,2 are back up, system 3 is still down.

As it's now 7am and system 3 is still down, notification window is now open:
"Warning system #3 went down outside of notification window and did not come back up, by 7am"

The long story short, is I monitor a lot of systems, many of which get a reboot once a fortnight for example at obscure times, I'd love to not be notified of these reboots.

### ❓ Alternatives

Nothing comes to mind, sorry.

### 📝 Additional Context

Dummy screenshot here:

[https://i.imgur.com/dMVeGfo.png](https://i.imgur.com/dMVeGfo.png)

Oh and thank you for the hard work as always, it's truly appreciated.

**Comment by CommanderStorm** [comment]
> If a system regularly goes down from say 09:00pm until 11pm, every second Saturday

Would you not be able to add this as a maintance window using cron?
![image](https://github.com/louislam/uptime-kuma/assets/26258709/25e71b0d-f707-48da-9d50-d9e7688a6f77)

**Comment by CommanderStorm** [comment]
Linking Related: https://github.com/louislam/uptime-kuma/issues/3792 https://github.com/louislam/uptime-kuma/issues/3404

**Comment by jaxjexjox** [comment]
Sorry I have no idea where you're configuring that?  I tried looking and I can't see it.

**Comment by jaxjexjox** [comment]
I believe this is potentially already handled by the app! thank you.

**Comment by CommanderStorm** [comment]
Bulk edit menus are tracked in https://github.com/louislam/uptime-kuma/issues/455 https://github.com/louislam/uptime-kuma/issues/2934 https://github.com/louislam/uptime-kuma/issues/1323

If you wan't to tackle some part of this work, here is our contribution guide https://github.com/louislam/uptime-kuma/blob/5b6522a54edad9737fccf195f9eaa25c6fb9d0f6/CONTRIBUTING.md

**Comment by jaxjexjox** [comment]
I just played with it, my only observation was, unless I did something wrong, I would love to just tick "all" (I didn't see it) instead of the 80+ systems I had to seleect.

It's probably there and I just didn't look.  
Thanks so much.
